responsibleai/tests/data_balance/conftest.py

```diff
@@ -0,0 +1,178 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+import copy
+
+import pandas as pd
+import pytest
+
+from ..common_utils import create_adult_income_dataset
+
+
+@pytest.fixture(scope="session")
+def adult_data():
+    (
+        data_train,
+        data_test,
+        y_train,
+        y_test,
+        categorical_features,
+        _,
+        target_col,
+        _,
+    ) = create_adult_income_dataset()
+    train_df = copy.deepcopy(data_train)
```

Comment:
You are fully right

I copied this fixture from the [causal conftest.py](https://github.com/ms-kashyap/responsible-ai-widgets/blob/7fd9b535e485ed9e104b7a3e238e877aa23a2e98/responsibleai/tests/causal/conftest.py#L16)

I couldn't use that fixture directly because the unit tests need other information such as categorical columns

I don't mind removing the `deepcopy()` for train and test. What are your thoughts?

Reply:
I realized maybe `conftest.py` in `causal` needs to do a deepcopy because it's a session-level fixture and therefore the fixture can mutate from test to test.

However, I ran data balance without the deepcopy and tests are running file. So I'll go with your preference!

Reply:
If it works without the deepcopy then I think you can remove it.  To me it looks like it isn't needed but I might be missing something.